core/sql/common/ComMisc.cpp

```diff
@@ -357,3 +358,119 @@ NABoolean ComTrafReservedColName(
 
   return FALSE;
 }
+
+
+Int32  ComGenerateUdrCachedLibName(NAString libname,Int64 redeftime, NAString schemaName, NAString userid, NAString &cachedLibName, NAString &cachedLibPath)
```

Comment:
Just to understand this function.  It creates the structure of directories (if not already created) and returns  the cached library name and cached library path.   If the user name is passed, then that username is used in the path.  If the user name is not passed, then that portion of the directory is set to DB__ROOT.   According the the header file, the username can also be public.  For the cached library name, we spilt the passed in name into two parts, the library name and the suffix (.so/.jar) and insert the redefinition timestamp in between.